### PR TITLE
feat: initial commit for erofs support

### PR DESCRIPTION
**Pull request by rchincha**

### CODE REVIEW

File: erofs/erofs.go [hunk 1]
```diff
@@ -0,0 +1,744 @@
...
+// previously included or excluded.
+type ExcludePaths struct {
+	exclude map[string]bool
+	include []string
+}
+
+type erofsFuseInfoStruct struct {
+	Path           string
+	Version        string
+	SupportsNotfiy bool
```
Comment (hallyn) [inline erofs/erofs.go]: This typo is consistent but will be murder to maintain later :)

File: erofs/erofs.go [hunk 2]
```diff
@@ -0,0 +1,744 @@
...
+}
+
+// ExcludePaths represents a list of paths to exclude in a erofs listing.
+// Users should do something like filepath.Walk() over the whole filesystem,
+// calling AddExclude() or AddInclude() based on whether they want to include
+// or exclude a particular file. Note that if e.g. /usr is excluded, then
+// everyting underneath is also implicitly excluded. The
+// AddExclude()/AddInclude() methods do the math to figure out what is the
+// correct set of things to exclude or include based on what paths have been
+// previously included or excluded.
```
Comment (hallyn) [inline erofs/erofs.go]: I'm finding this hard to reason about, and therefore finding it hard to figure out whether AddExclude() is doing the right thing.

I think you're mainly thinking in terms of "I am generating a changeset and therefore want to include only files which have changed", is that right?

File: erofs/erofs.go [hunk 3]
```diff
@@ -0,0 +1,744 @@
...
+		include: []string{},
+	}
+}
+
+func (eps *ExcludePaths) AddExclude(p string) {
+	for _, inc := range eps.include {
+		// If /usr/bin/ls has changed but /usr hasn't, we don't want to list
+		// /usr in the include paths any more, so let's be sure to only
+		// add things which aren't prefixes.
+		if strings.HasPrefix(inc, p) {
```
Comment (hallyn) [inline erofs/erofs.go]: But what if both /usr/bin/cal and /usr/bin/calendar have changed?  Will this check end up possibly ignoring /usr/bin/cal?

File: erofs/erofs.go [hunk 4]
```diff
@@ -0,0 +1,744 @@
...
+
+	blob, err := os.Open(tmpErofs.Name())
+	if err != nil {
+		return nil, "", rootHash, errors.WithStack(err)
+	}
+
+	return blob, GenerateErofsMediaType(compression, verity), rootHash, nil
+}
+
+func isMountedAtDir(_, dest string) (bool, error) {
```
Comment (hallyn) [inline erofs/erofs.go]: Is checking that the source is correct a TODO?

**Comment by rchincha** [comment]
PR #25 supersedes this one, so closing.

**Pull request closed without merging**